recipes/logfault/all/conanfile.py

```diff
@@ -0,0 +1,86 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.files import copy, get
+from conan.tools.layout import basic_layout
+from conan.tools.scm import Version
+from conan.tools.cmake import CMake, CMakeToolchain
+import os
+
+
+required_conan_version = ">=1.52.0"
+
+
+class PackageConan(ConanFile):
+    name = "logfault"
+    description = "Simple to use, header only C++ library for application-logging on all major platforms."
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/jgaa/logfault"
+    topics = ("logging", "header-only")
+    package_type = "header-library"
+    settings = "os", "arch", "compiler", "build_type"
+    no_copy_source = True
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+           "Visual Studio": "15",
+           "msvc": "191",
+           "gcc": "8",
+           "clang": "7",
+           "apple-clang": "12",
+        }
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables['LOGFAULT_BUILD_TESTS'] = not self.conf.get("tools.build:skip_test", default=True, check_type=bool)
+        tc.generate()
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        if not self.conf.get("tools.build:skip_test", default=True):
+            self.requires("gtest/1.14.0")
+
+    def package_id(self):
+        self.info.clear()
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def build(self):
+        if not self.conf.get("tools.build:skip_test", default=True):
+            cmake = CMake(self)
+            cmake.configure()
+            cmake.build()
+            cmake.test()
```

Comment:
```suggestion
        cmake = CMake(self)
        cmake.configure()
        cmake.build()
```
Running tests, even optionally, is discouraged on CCI.